regular_expressions_groups/regular_expressions_groups.md

```diff
@@ -0,0 +1,280 @@
+
+### Capturing Groups in a Regex Checker
+
+You can look at the capturing results for multiple groups and practice this skill in a regex checker, regardless of whether you also use a programming language like R or Python.  For example, let's say we wanted to extract all four components of our long id, `002923_P_A56G199C2_030023472948651`. 
+
+We can put parentheses around each group and add this regular expression to a regex checker like [Regex 101](https://www.regex101.com) and check to make sure we're capturing what we intend to.
+
+We'll add the pattern `(\d+)_([PMF])_([A-Z\d]+)_(\d+)` to the small box labeled "regular expression", and our sample id, `002923_P_A56G199C2_030023472948651`, to the larger box labeled "test string".  [We can then see our captured groups on the right!](https://regex101.com/r/XmG4kp/1)
+
+![The regex101 website with the aforementioned pattern and test string added.  Four groups are indicated in colorful highlights, listed in the match information pane, and explained in depth in the explanation pane.](media/capturing_groups.png)<!-- style = "border: 1px solid rgb(var(--color-highlight));" -->
+
+We see four groups listed in the "Match Information" pane on the right side.  Above that pane, the "Explanation" pane goes into detail about each group.
+
+
+### Quiz: Capturing Groups
+
+Consider a regular expression with capturing group that describes email addresses:
+
+`[A-Za-z\d\-\._]+@([A-Za-z\d\-\._]+\.[A-Za-z]{2,})`
+
+ What will the capturing group in this regular expression actually capture?
+
+[( )] The entire email address
+[( )] The user mailbox (the entire part before the `@` of the email address)
+[(X)] The email domain (the entire part following the `@` of the email address)
+[( )] The TLD, or "top level domain" (the last period and last few letters, like `.com`, `.edu`, or `.gov`)
+[( )] None of these
+[[?]] Hint: Feeling stuck?  Try using [Regex 101](https://www.regex101.com)!
+***
+<div class = "answer">
+
+This regex looks intimidating, but it's really only a few parts.  Let's break it apart considering the email address `regex_is_awesome_123@fakemail.org`
+
+* `[A-Za-z\d\-\._]+` means one or more letter, number, hyphen, period, or underscore (in our case, "regex\_is\_awesome\_123")
+* `@` indicates the `@`
+* `[A-Za-z\d\-\._]+` again means one or more letter, number, hyphen, period, or underscore ("fakemail")
+* `\.` indicates a literal period
+* `[A-Za-z]{2,}` indicates two or more letters ("org").
+
+In our case, we had a capturing group around the last three of those bullet points.  That means that what is captured is "fastmail.org", the entire email domain.
```

Comment:
```suggestion
In our case, we had a capturing group around the last three of those bullet points.  That means that what is captured is the entire email domain ("fakemail.org").
```
I think this matches the style of the previous examples in this section and solves a bit of my punctuation/quotation/verbatim angst.